### PR TITLE
chore(policies): use display-name label for policy listing

### PR DESCRIPTION
**Pull request by johncowen**
Use `display-name` and `namespace` labels for policy listings

---

I noticed that the mock for the affected dataplanes endpoint was static, so I changed that up but also that this endpoint is paged, but we don't offer any paging and we use frontend searching on the single page. I stopped short of doing anything towards changing this tho. Not sure if its on purpose or just an accidental omission? We might want to follow up with changing that soon.

Closes https://github.com/kumahq/kuma-gui/issues/2372

### CODE REVIEW

File: src/types/index.d.ts [hunk 1]
```diff
@@ -390,7 +390,10 @@ export interface DataPlane extends MeshEntity {
  */
 export interface DataPlaneOverview extends MeshEntity {
   type: 'DataplaneOverview'
-  labels?: Record<string, string>
+  labels?: {
+    'kuma.io/display-name'?: string
+    'k8s.kuma.io/namespace'?: string
```
Comment (kleinfreund) [inline src/types/index.d.ts:395]: **Suggestion**: This change would prevent accessing other properties. It should be:

```suggestion
    'k8s.kuma.io/namespace'?: string
    [key: string]: string | undefined
```

Comment (johncowen) [reply src/types/index.d.ts:395]: There's something to be said for not letting ourselves statically accessing properties that aren't specified here but 🤷

**Review by kleinfreund — APPROVED** [review]

**Comment by johncowen** [comment]
> but we don't offer any paging and we use frontend searching on the single page. I stopped short of doing anything towards changing this tho. Not sure if its on purpose or just an accidental omission?

@kleinfreund any thoughts on this ^

**Comment by kleinfreund** [comment]
> @kleinfreund any thoughts on this ^

Not really. This can definitely be improved, but users are relatively unlikely to run into limitations here

**Pull request merged**